examples/js/archive.js

```diff
@@ -73,13 +76,65 @@ function renderImages(file, url) {
 
     placeButton.classList.add('btn', 'btn-sm', 'btn-outline-secondary', 'place-button');
     placeButton.innerHTML = 'Place on map';
-
+    image.setAttribute('data-original-image', `${url.replace('metadata', 'download')}/${file.name}`);
     image.src = `${url.replace('metadata', 'download')}/${file.name}`;
     imageRow.classList.add('d-flex', 'justify-content-between', 'align-items-center', 'mb-4', 'pe-5');
     imageRow.append(image, placeButton);
+    imageContainer.appendChild(imageRow);
+    imageCount++;
+  });
+};
+
+// this function is called when there are more than a 100 images and thumbnails in data fetched and also if there are not enough thumbnails per image
+// it takes in three arguments thumbs, url and imgs
+// using defualt function parameters and a logical OR, it is reuseable as a graceful failsafe for when there are no thumbnails for all images.
+const getThumbs = (thumbs = [], url, imgs) => {
+  const display = thumbs || imgs;
+
+  display.forEach((file) => {
+    const imageRow = document.createElement('div');
+    const image = new Image(65, 65);
+    const placeButton = document.createElement('a');
+    fetchedFrom = document.createElement('p');
```

Comment:
```suggestion
    // link back to the images' source URL
    fetchedFrom = document.createElement('p');
```

Reply:
Just noting that a few very brief explanatory comments can help break up and explain a longer block of code like this. But if we name the variables very specifically, sometimes we can even skip this! Think of it like reading a short story. Is it clear to another reader what's happening in each section?

Reply:
Hmm, Thank you for the insight, I would definitely take it into consideration henceforth